crypto/x509/x_pubkey.c

```diff
@@ -147,11 +149,21 @@ static int x509_pubkey_ex_d2i_ex(ASN1_VALUE **pval,
         return 0;
     }
 
+    tmppropq = ((X509_PUBKEY *)*pval)->propq;
+
     /* This ensures that |*in| advances properly no matter what */
     if ((ret = ASN1_item_ex_d2i(pval, in, len,
```

Comment:
I would propose an alternative solution. `ASN1_item_ex_d2i` is just a wrapper around `asn1_item_ex_d2i_intern` which does this:

https://github.com/openssl/openssl/blob/85a8eba56769c4d25a66e0b52e8fb3e76bbe4afe/crypto/asn1/tasn_dec.c#L107-L123

The NULL checks at the beginning of this function are actually redundant since `asn1_item_embed_d2i` repeats them. So the only thing this function adds on top of `asn1_item_embed_d2i` is the freeing functionality on failure. I suggest making  `asn1_item_embed_d2i` non static, and adding its declaration to `include/internal/asn1.h`. We can then call it directly here and avoid the freeing behaviour. If it fails (return value <=0) then we can free `pval` here with a call to `x509_pubkey_ex_free`.